--- a/src/libFLAC/stream_encoder.c
+++ b/src/libFLAC/stream_encoder.c
@@ -1883,6 +1883,19 @@
 	ok &= FLAC__stream_encoder_set_do_mid_side_stereo          (encoder, compression_levels_[value].do_mid_side_stereo);
 	ok &= FLAC__stream_encoder_set_loose_mid_side_stereo       (encoder, compression_levels_[value].loose_mid_side_stereo);
 #ifndef FLAC__INTEGER_ONLY_LIBRARY
+#if defined(CODEC_STREAM_ONLY)
+	encoder->protected_->num_apodizations = 1;
+	if(value >= 6) {
+		const uint32_t parts = value >= 8 ? 3 : 2;
+		encoder->protected_->apodizations[0].type = FLAC__APODIZATION_SUBDIVIDE_TUKEY;
+		encoder->protected_->apodizations[0].parameters.subdivide_tukey.parts = parts;
+		encoder->protected_->apodizations[0].parameters.subdivide_tukey.p = 0.5 / parts;
+	}
+	else {
+		encoder->protected_->apodizations[0].type = FLAC__APODIZATION_TUKEY;
+		encoder->protected_->apodizations[0].parameters.tukey.p = 0.5;
+	}
+#else
 #if 1
 	ok &= FLAC__stream_encoder_set_apodization                 (encoder, compression_levels_[value].apodization);
 #else
@@ -1892,6 +1905,7 @@
 	encoder->protected_->apodizations[0].parameters.tukey.p = 0.5;
 #endif
 #endif
+#endif
 	ok &= FLAC__stream_encoder_set_max_lpc_order               (encoder, compression_levels_[value].max_lpc_order);
 	ok &= FLAC__stream_encoder_set_qlp_coeff_precision         (encoder, compression_levels_[value].qlp_coeff_precision);
 	ok &= FLAC__stream_encoder_set_do_qlp_coeff_prec_search    (encoder, compression_levels_[value].do_qlp_coeff_prec_search);
